e2e/README.md

```diff
@@ -145,7 +149,1089 @@ func E2ETests(env e2e.TestEnv) testhelper.Tests {
+file (e.g. `imgbuild.go`), and now... your test doesn't seem to be running. You
+can find it in the e2e output logs anywhere.
+
+**Common cause:** You've forgotten to add your test function to the
+`testhelper.Tests` map created by your testing group's E2ETests() function.
+
+This function is typically located at the bottom of the Go source file
+corresponding to the e2e group (see the [introduction](#introduction) for an
+example). Your test function - really, a method of the group's e2e context
+object - won't run unless it is added to the `testhelper.Tests` map.
+
+Make sure to also [mark your test
+appropriately](#parallel-par-vs-non-parallel-seq-tests) if it cannot be run in
+parallel.
+
+**Scenario 2:** You're trying to get the e2e test to only run your test. You've
+set the [E2E_GROUPS and E2E_TESTS](#running-the-e2e-suite) environment
+variables, and you've double checked the spelling of the group & test names, but
+it's not catching your test; nothing is running.
+
+**Common cause:** You're trying to use E2E_TESTS to filter tests by something
+*other* than the top-level test name. Here is an example of a test name from a
+specific e2e subtest:
+
+```plain
+TestE2E/SEQ/INSTANCE/auth/Root/start_before_auth
+```
+
+`TestE2E` is the single top-level name for all e2e tests; next comes [`PAR` or
+`SEQ`](#parallel-par-vs-non-parallel-seq-tests), then comes the group name
+(which is what E2E_GROUPS matches), followed by the top-level test name (which
+is what E2E_TESTS matches; in this case, `auth`). The E2E_TESTS environment
+variable is a regular expression that is matched only against the top-level test
+name.
+
+At the present, there is no way to match against more deeply-embedded components
+of the test path. However, individual top-level tests are usually short enough
+to make this granularity sufficient in practice.
+
+**Another common cause:** You've confused the test function's name for the test
+name, or the test group source file's name for the group name.
+
+The values that E2E_TESTS matches against are the values of the keys in the
+`testhelper.Tests` map. Consider this example, repeated from earlier:
+
+```go
+func E2ETests(env e2e.TestEnv) testhelper.Tests {
+	c := ctx{
+		env: env,
+	}
+
+	np := testhelper.NoParallel
+
+	return testhelper.Tests{
+		"add":            c.remoteAdd,
+		"list":           c.remoteList,
+		"default or not": c.remoteDefaultOrNot,
+		"remove":         c.remoteRemove,
+		"status":         c.remoteStatus,
+		"test help":      c.remoteTestHelp,
+		"use":            c.remoteUse,
+		"use exclusive":  np(c.remoteUseExclusive),
+	}
+}
+```
+
+To run only the c.remoteUseExclusive() test, the environment variable value
+`E2E_TESTS=remoteUseExclusive` won't work. That's because, as far as the testing
+suite is concerned, the test's name is `use exclusive`. Therefore, a correct
+value to run this test only would be, e.g., `E2E_TESTS=use.exclusive`.
```

Comment:
```suggestion
value to run this test only would be, e.g., `E2E_TESTS=use exclusive`.
```

Reply:
What's wrong with using a dot here...? Note that, without the dot, you'd have to quote the value in some contexts (e.g. when using the prefixed command line pattern, as in `E2E_TESTS=use.exclusive make -C ...`, whereas with the dot, you don't.

Reply:
Using the `.` without explaining how it works is inviting people to think that e.g. Go replaces spaces in test names with `.`

I'd prefer showing the quoting, as that's correct for matching only that test name. Otherwise please mention why the `.` works.